DarwinWorld/src/main/java/agh/ics/oop/model/mapElements/Animal.java

```diff
@@ -78,9 +80,14 @@ public void drawOnMap(GraphicsContext gc, double fieldWidth, double fieldHeight)
     }
 
     public void reproduction(Animal child) {
-        this.energy-=reproductionCost;
+        this.energy -= reproductionCost;
+        if (alfaAnimal != null) {
+            alfaAnimal.descendants += 1;
```

Comment:
Tutaj temu samemu dziecku dwa razy ustawiamy alfaanimal, nie lepiej zrobić to w mapfield ?

Reply:
ten if właśnie zapobiega podwójnemu ustawianiu alfaanimal.
gdyby wyciągnąć tą funkcjonalność do poziomu map field to trzeba by było sprawdzać dla każdego parenta czy ma alfaanimala  
![Zrzut ekranu 2024-01-16 o 9 26 51 AM](https://github.com/Wenszel/darwin-world/assets/46707987/ecc1b792-7ed2-4c20-923a-7ec39eff0b8b)

Reply:
A to jest wrzucone już ? bo nie widzę tego kodu nigdzie.

Reply:
to napisałem na szybko lokalnie, żeby sprawdzić jakby to wyglądało